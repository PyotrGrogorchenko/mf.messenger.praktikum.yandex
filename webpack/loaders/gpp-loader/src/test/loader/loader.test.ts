import 'core-js/stable'
import 'regenerator-runtime/runtime'

import compiler from './compiler'

test('Loader component Bb', async () => {
  const stats = await compiler('components/Bb/Bb.gpp.ts', { componentsDir: 'components' })
  const statsJson = stats.toJson({ source: true })
  if (statsJson.modules) {
    const output = statsJson.modules[0].source
    expect(output).toMatchSnapshot()
  }
})

test('Loader index', async () => {
  const stats = await compiler('Index.gpp.ts', { componentsDir: 'components' })
  const statsJson = stats.toJson({ source: true })
  if (statsJson.modules) {
    const output = statsJson.modules[0].source
    expect(output).toMatchSnapshot()
  }
})
