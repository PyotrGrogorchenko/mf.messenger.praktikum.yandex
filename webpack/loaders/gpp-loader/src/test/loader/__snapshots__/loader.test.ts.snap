// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader component Bb 1`] = `
"//#gpp-loader #importComponents\\\\\\\\
import {Cc} from '/home/pyotr/prg/mf.messenger.praktikum.yandex/webpack/loaders/gpp-loader/src/test/loader/example/components/CD/Cc/Cc.gpp.ts'
import {Dd} from '/home/pyotr/prg/mf.messenger.praktikum.yandex/webpack/loaders/gpp-loader/src/test/loader/example/components/CD/Dd/index.ts'
//#gpp-loader #importComponents//
export class Bb {
  //#gpp-loader #componentsfunc\\\\\\\\
components() {return {Cc,Dd}}
//#gpp-loader #componentsfunc//
template() {
    \`<Cc>
      <h1>ClassC</h1>
    </Cc>
    <Dd>
      <h1>ClassD</h1>
    </Dd>\`
  }
}"
`;

exports[`Loader index 1`] = `
"//#gpp-loader #importComponents\\\\\\\\
import {Aa} from '/home/pyotr/prg/mf.messenger.praktikum.yandex/webpack/loaders/gpp-loader/src/test/loader/example/components/Aa/Aa.gpp.ts'
import {Dd} from '/home/pyotr/prg/mf.messenger.praktikum.yandex/webpack/loaders/gpp-loader/src/test/loader/example/components/CD/Dd/index.ts'
//#gpp-loader #importComponents//
export class Index {
  //#gpp-loader #componentsfunc\\\\\\\\
components() {return {Aa,Dd}}
//#gpp-loader #componentsfunc//
template() {
    \`<Aa>
      <h1>ClassA</h1>
      <Dd>
        <h1>ClassB</h1>
      </Dd>
    </Aa>\`
  }
}
"
`;
